contracts/reserve/GoodReserveCDai.sol

```diff
@@ -306,36 +464,37 @@ contract GoodReserveCDai is
 					getMarketMaker(),
 					this,
 					msg.sender,
-					_sellTo,
-					_gdAmount.sub(discount)
+					sellParameters.sellTo,
+					sellParameters.gdAmount.sub(discount)
 				);
 
 		uint256 tokenReturn =
 			getMarketMaker().sellWithContribution(
-				_sellTo,
-				_gdAmount,
+				sellParameters.sellTo,
+				sellParameters.gdAmount,
 				contributionAmount
-			);
-		require(
-			tokenReturn >= _minReturn,
-			"Token return must be above the minReturn"
 		);
 		require(
-			_sellTo.transfer(msg.sender, tokenReturn) == true,
-			"Transfer failed"
-		);
-
-		emit TokenSold(
-			msg.sender,
-			address(_sellTo),
-			_gdAmount,
-			contributionAmount,
-			_minReturn,
-			tokenReturn
+			tokenReturn >= sellParameters.minReturn,
+			"Token return must be above the minReturn"
 		);
-		return tokenReturn;
+		if(_returnInCDai == true){
+			
+			return (tokenReturn, contributionAmount);
+
+		} else {
+			cERC20 cDai = cERC20(nameService.getAddress("CDAI"));
+			ERC20 dai = ERC20(nameService.getAddress("DAI"));
+			
+			uint256 currDaiBalance = dai.balanceOf(address(this));
+			
+			uint256 daiResult = cDai.redeem(tokenReturn);
+			uint256 daiReturnAmount = (dai.balanceOf(address(this))).sub(currDaiBalance);
+			
+			return (daiReturnAmount, contributionAmount);
```

Comment:
i've already requested return should always be in cDAI
handle conversion in outer method